From 6b6294302924b24f8b30b080303fbd306c13b913 Mon Sep 17 00:00:00 2001
From: Mihai Serb <mihai.serb@freescale.com>
Date: Mon, 12 Mar 2012 21:50:25 +0000
Subject: [PATCH 114/518] Fix bug in inbound policy verification table
 configuration

Because of a typo, the action for a frame that fails inbound policy
verification was configured to always drop frames. The frames should
be enqueued to the FQ provided explicitly for this purpose.

Signed-off-by: Mihai Serb <mihai.serb@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_ipsec.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
index e02a8d0..fe1ac6a 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_ipsec.c
@@ -2634,7 +2634,7 @@ int dpa_ipsec_create_sa(int dpa_ipsec_id,
 			}
 
 			memset(&miss_action, 0, sizeof(miss_action));
-			fill_cls_action_enq(&action, FALSE,
+			fill_cls_action_enq(&miss_action, FALSE,
 				sa->policy_miss_fqid, NULL);
 
 			err = dpa_classif_table_modify_miss_action(inbpol_td,
-- 
1.7.5.4

